apps/settings/js/panels/simcard_manager/simcard_manager.js

```diff
@@ -152,15 +151,16 @@ define(function(require) {
         if (prevCardIndex !== newCardIndex) {
           // UX needs additional hint for users to make sure
           // they really want to change data connection
-          var wantToChange =
-            window.confirm(_('change-outgoing-data-confirm'));
+          l10n.formatValue('change-outgoing-data-confirm').then(msg => {
+            var wantToChange = window.confirm(msg);
 
-          if (wantToChange) {
-            SimSettingsHelper.setServiceOnCard('outgoingData',
-              newCardIndex);
-          } else {
-            this.selectedIndex = prevCardIndex;
-          }
+            if (wantToChange) {
+              SimSettingsHelper.setServiceOnCard('outgoingData',
+                newCardIndex);
+            } else {
+              this.selectedIndex = prevCardIndex;
+            }
+          });
```

Comment:
Here it seems to work well, because the event is asynchronous anyway (reacts to blur).